### PR TITLE
Implement static wait for

### LINKED ISSUE #847: `Electron.wait_for` in a workflow causes the workflow function to no longer be callable
The new `Electron.wait_for()` method allows users to instruct Covalent that a task (task B) should wait for another task (task A) to return before being dispatched. This is needed in some instances because Covalent normally infers the dependency relations between electrons just from their inputs and outputs and therefore would not be able to handle scenarios where task B depends only on a side-effect of task A.

However, when users construct a function describing their workflow, invoking the `wait_for()` method on an electron causes the workflow function to no longer be callable:
```python

@ct.lattice
def workflow_function():
    res1 = task_1() # task_1 is an electron which returns an int at runtime
    res2 = task_2() # evaluates to an int at runtime
    res2.wait_for(res1) # int doesn't have a wait_for method
```
When `workflow_function` is called, `res1` and `res2` are the return values of `task_1` and `task_2`, not Electrons, so Python function cannot evaluate the third statement. 

This behavior is unintuitive because workflow functions are Python *functions*, and functions are expected to be callable. The `ct.lattice` decorator is not relevant here because:
* Users may construct their workflow function first as an ordinary function, and then construct a lattice explicitly using `ct.lattice()`, [as we demonstrate](https://covalent.readthedocs.io/en/latest/tutorials/machine_learning/mnist_classifier.html#Transform-and-run-workflow-with-Covalent) in the MNIST tutorial.
* Decorators are expected to merely [extend the behavior of a function](https://realpython.com/primer-on-python-decorators/). 

**Suggestion**: Instead of exposing the `Electron.wait_for` method directly to users, we can treat that a private method like the other methods that manipulate the transport graph (e.g. `Electron.connect_node_with_others()`), and expose a static wrapper function that invokes `Electron.wait_for` only in the context of an active lattice (during graph building and postprocessing). For example:

```python
def wait(child, parents):
    active_lattice = active_lattice_manager.get_active_lattice()

    if active_lattice:
        return child.wait_for(parents)
    else:
        return child
```
Then the following modified workflow function can both be called and dispatched:

```python
@ct.lattice
def workflow_function():
    res1 = task_1()
    res2 = task_2() 
    ct.wait(child=res2, parents=res1)
```

### PR DESCRIPTION
**Pull request by cjao**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

Closes #847.

### CODE REVIEW

File: doc/source/concepts/concepts.rst [hunk 1]
```diff
@@ -183,7 +183,11 @@ This ensures that the independent experiments are performed in parallel rather t
 Waiting for other electrons
 ----------------------
 
-Sometimes the user might want to wait for a task's execution before executing another task even when the output of one is not the input of another. For those cases a `wait_for()` function is there which is called on the dependent electron and takes as input the depending electrons,
+Covalent normally infers the dependencies between electrons from
+their inputs and ouputs. Sometimes the user might want to wait for a
+task's execution before executing another task even when the output of
+one is not the input of another. The `wait()` function can come in
+handy in those cases.
```
Comment (venkatBala) [inline doc/source/concepts/concepts.rst:190]: The `wait()` function can come handy in such cases?

**Review by venkatBala — APPROVED** [review]
looks good... small PR

File: doc/source/concepts/concepts.rst (hunk 1 above)
Comment (cjao) [reply doc/source/concepts/concepts.rst:190]: Good catch, thanks.

**Comment by cjao** [comment]
I'm leaving this as draft for now since it involves a (small) user-facing API change. @santoshkumarradha.

File: covalent/_workflow/electron.py [hunk 2]
```diff
@@ -602,3 +602,26 @@ def wrapper(*args, **kwargs):
...
+    Useful when execution of an electron relies on a side-effect
+    from another one.
+
+    """
+    active_lattice = active_lattice_manager.get_active_lattice()
+
+    if active_lattice:
+        return waiting_electron.wait_for(parents)
+    else:
+        return waiting_electron
```
[the remark this answers is not in the document]
Comment (cjao) [reply covalent/_workflow/electron.py:627]: Good point. Or would you suggest making it a static method of `Electron`?

[the remark this answers is not in the document]
Comment (cjao) [reply covalent/_workflow/electron.py:627]: One potential problem is that `electron.py` already imports `utils.py`, so to reference Electrons from `utils.py` would create a circular import.

[the remark this answers is not in the document]
Comment (kessler-frost) [reply covalent/_workflow/electron.py:627]: Where are we referencing `Electron` in this function? If you're talking about typehint then you can use `if TYPE_CHECKING:` and enclose `Electron` as `"Electron"` and typehinting will work fine.

**Comment by wjcunningham7** [comment]
This pull request is stale because it has been in review for at least three days. It will be closed in two more days.

**Review by santoshkumarradha — DISMISSED** [review]

**Review by kessler-frost — APPROVED** [review]

File: covalent/_workflow/electron.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cjao) [reply covalent/_workflow/electron.py]: `wait` is an operation on Electrons and is not used anywhere else.

**Review by santoshkumarradha — APPROVED** [review]

**Pull request merged**